contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -4968,9 +4991,15 @@ transform_pivot_clause(ParseState *pstate, SelectStmt *stmt)
 	pivot_src_sql->groupClause = copyObject(src_sql_groupbylist);
 	pivot_src_sql->sortClause = copyObject(src_sql_sortbylist);
 	
-	/* use the copy of the orgininal fromClause to prevent double analyzing fromClause */
+	/* use the copy of the orgininal fromClause and withClause to prevent double analyzing fromClause */
 	((SelectStmt *)stmt->srcSql)->fromClause = src_sql_fromClause_copy;
-	/* Transform the new src_sql & get the output type of that agg function*/
+	((SelectStmt *)stmt->srcSql)->withClause = with_clause;
+
+	/* we rewrite srcSql object refereces again because we used a new copy of fromClause */
+	if (enable_schema_mapping())
+		rewrite_object_refs((Node *)stmt->srcSql);
```

Comment:
Suggestion: lets add a test case where all objects are completely qualified.
for example babelfish allows db.schema.table.colname & db.schema.tablename